changes/2332.feature

```diff
@@ -0,0 +1 @@
+Build http-parser extension in non-strict mode by default.
```

Comment:
I'm not sure we want non-strict mode by default.

Reply:
Also, does this switch implies that we have to ship two wheels with strict and non strict parsers?

Reply:
Why not?
We are strict on sending HTTP data but relaxed on receiving the data from foreign servers.
Basically we do the same in very many places.

Reply:
According the sources, this check does a bit more than just allow non-ascii characters in URL. If I read correctly, users may send "GET / OLOLO/1.1" requests without worry about, servers may return malformed status codes, etc. I don't think we want to support this.

Reply:
`GET / OLOLO/1.1` does not work — have just checked this.

Reply:
Hm, then I read it wrong way. What exact it controls then? Can't found any documentation about.

Reply:
I was looking at tests: https://github.com/nodejs/http-parser/blob/05525c5fde1fc562481f6ae08fa7056185325daf/test.c#L766-L811

Reply:
After brief overview I've found the following behavior changes for non-strict mode.

It allows 
1. spaces, tabs and linefeeds in headers
2. utf-8 in URLs
3. underscore in hostname
4. `\9` (ht) and `\12` (np) symbols. 

Also looks like it has relaxed checks for `\r\n` (`\r` and `\n` are allowed).
"GET / OLOLO/1.1" if forbidden but "GET / HLOL/1.1" is allowed.

Reply:
@asvetlov Thanks! 

Are we ok with 1, 3 and 4?

Reply:
I think yes. It doesn't raise security issues IMHO